Jenkins_setup.md

```diff
@@ -0,0 +1,161 @@
+> still test for the GHPRB varibles to distinguish between a PR and Master build.
+
+This script is placed into the Jenkins build dialog like:
+
+![Jenkins Build dialog](pictures/jenkins_build.png)
+
+Teardown script:
+
+```bash
+#!/bin/bash
+
+export GOPATH=$WORKSPACE/go
+
+cd $GOPATH/src/github.com/kata-containers/tests
+.ci/teardown.sh "$WORKSPACE"
+```
+
+The teardown script is entered into the Jenkins Post-Build Actions dialog like:
+
+![Jenkins Build dialog](pictures/jenkins_build.png)
+
+## Environment variables
+
+The `jenkins_job_build.sh` script both checks for some variables to decide what to do,
+and sets some variables for sub-scripts to check.
+
+### Input variables
+
+The following variables influence the actions of the script:
+
+| Var                 | Effect                                                           |
+| ---------           | ---------------------------------------------------------------- |
+| `WORKSPACE`         | Must be set. Is set by Jenkins to point at the directory the script will use as its workspace (scratch area) |
+| `ghprbTargetBranch` | Is set by the GHPRB Jenkins plugin. Is only set for PR builds, and is used with `ghprbPullId` to distinguish between PR and Master builds. |
+| `ghprbPullId`       | Is used in PR builds to locate the correct branch on github to pull and build. |
+| `KATA_DEV_MODE`     | Must not be set, or the script will not act as a CI script. KATA_DEV_MODE is a developer only option allowing the script to be run 'safely' in a development environment. |
+| `METRICS_CI`        | If set, will skip running the QA tests after the install/build phases (as the tests are not required for a metrics run). |
+
+### Output variables
+
+| Var                 | Effect                                                           |
+| ---------           | ---------------------------------------------------------------- |
+| `CI`                | Is set by the script (unless KATA_DEV_MODE is already set) to indicate to sub-scripts that this is a CI build. |
+| `GOPATH`            | Is set to the installed Go environment                           |
+| `GOROOT`            | Is set to the installed Go environment                           |
+| `PATH`              | Is modified to include ${GOPATH}/bin, /usr/local/go/bin and the 'sbin' directories. |
+| `kata_repo`         | Is set to the github repo name, as passed in to $1 of the script. |
+
+## Experienced users
+
+For the more experienced Jenkins user, if you are au-fait reading Jenkins xml config files
+then you can find a backup of our Azure based Jenkins QA CI in this repository under the
+[jenkins sub folder](https://github.com/kata-containers/ci/tree/master/jenkins).
+
+## Other CIs
+
+Formerly Kata Containers, and one of its predecessors [Clear Containers](https://github.com/clearcontainers)
+have used both [Travis](https://travis-ci.com/) and [SemaphoreCI](https://semaphoreci.com/)
```

Comment:
used both [Travis](https://travis-ci.com/) and [SemaphoreCI](https://semaphoreci.com/)

Reply:
hi @jodh-intel @grahamwhaley , where does jenkins_job_build.sh run into, in the $JENKINS_HOME or in jenkins web configure UI?
I think the steps to set up jenkins ci system in a new machine is :  1. install jenkins 2. add token and credential to connect github 3 add jobs.  is that right? when to run jenkins_job_build.sh?